smartcontractkit/operator-ui
# fix: release when using node 18.19

## Description

[`node 18.19`](https://nodejs.org/en/blog/release/v18.19.0) broke `ts-node` when using `.mts` file extensions (see: https://github.com/TypeStrong/ts-node/issues/2094). The release job hasn't been run since before the release of `18.19`, and because the `setup-node` action hasn't been pinned to a specific minor release of `node18`.

This was found because #62 was merged and the workflow experienced an error.

```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".mts" for /home/runner/work/operator-ui/operator-ui/scripts/release.mts
```

This PR:
1. Uses `node --loader=ts-node/esm ...` instead of invoking `ts-node` directly
2. Updates `.tool-versions` to use `nodejs 18.19.0`
3. Bumps `action/setup-node` to use the latest release, and uses `node-version-file` instead of `node-version`
4. Bumps `action/checkout` to use the latest release

## Steps to Test

1. Add `--draft` to `scripts/release.mts` to ensure you don't create a github release
```
await $`gh release create --draft ...`
```
2.  While in `scripts` directory - `yarn release`


# Checklist

If this PR creates changes to the operator-ui itself, rather than tests, pipeline changes, etc. Then please create a changeset so that a new release is created, and the changelog is updated. See: https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md#what-is-a-changeset
